cloudformation.yaml

```diff
@@ -343,14 +366,19 @@ Resources:
                "ClusterSizeCheck": {
                  "Type": "Task",
                  "Resource": "${ClusterSizeCheckArn}",
-                 "Next": "MigrateShards",
+                 "Next": "ReattachOldInstance",
                  "Retry": [ {
                    "ErrorEquals": [ "States.ALL" ],
                    "IntervalSeconds": 30,
                    "MaxAttempts": 20,
                    "BackoffRate": 1.0
                   } ]
                },
+               "ReattachOldInstance": {
```

Comment:
I think it would be safe to retry this task on error if we wanted to? (That's probably true for some existing lambdas too, so I am happy to review that as part of a separate change if you'd prefer to keep this one simple)

Reply:
I think that if we made this lambda retry, we might have to add some more stuff to the lambda: to check before we re-attempt the reattach, that the instance has not already been reattached! Otherwise, we might be attempting to reattach an instance that is already in the ASG, and this could cause the `attach` call to fail indefinitely?

I'll merge this for now if that's alright- we can always come back to it.

Reply:
Yes good point, retries always add a certain amount of complexity so I agree that it's better to revisit this later if necessary 👍